packages/gatsby-theme-docs/src/components/link.js

```diff
@@ -0,0 +1,237 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Location } from '@reach/router';
+import { Link as GatsbyLink, withPrefix } from 'gatsby';
+import { css } from '@emotion/core';
+import styled from '@emotion/styled';
+import { OutboundLink } from 'gatsby-plugin-google-analytics';
+import ExternalLinkIcon from '../icons/external-link-icon.svg';
+import { colors, dimensions } from '../design-system';
+import { useSiteData } from '../hooks/use-site-data';
+
+const isProduction = process.env.NODE_ENV === 'production';
+
+const productionHostname = 'docs.commercetools.com';
+const dummyHostname = 'dummy.com';
+
+// Gatsby links should be written with the full url path starting with `/` (so no
+// "filesystem"-relative paths).
+// In order to resolve relative paths that do no start with a `/`, we need to first
+// trim the possible `pathPrefix` included in the `location` of `@react/router`.
+const withoutPrefix = (value, pathPrefix) =>
+  value.replace(new RegExp(`^${pathPrefix}`), '');
+
+const linkStyles = css`
+  text-decoration: underline;
+  &,
+  > code {
+    color: ${colors.light.link};
+    :active,
+    :focus,
+    :hover {
+      color: ${colors.light.linkHover};
+    }
+  }
+`;
+
+export const ExternalSiteLink = props => (
+  <OutboundLink
+    {...props}
+    css={linkStyles}
+    target="_blank"
+    rel="noopener noreferrer"
+  />
+);
+const AnchorLink = styled.a`
+  ${linkStyles}
+`;
+const InternalSiteLink = styled(AnchorLink)``;
+const HistoryLink = styled(GatsbyLink)`
+  ${linkStyles}
+`;
+const InlineLink = styled.span`
+  display: inline-flex;
+  align-items: center;
+  > * + * {
+    margin: 0 0 0 ${dimensions.spacings.xs};
+  }
+  svg {
+    * {
+      fill: ${colors.light.link};
+    }
+  }
+  :hover {
+    svg {
+      * {
+        fill: ${colors.light.linkHover};
+      }
+    }
+  }
+`;
+
+/**
+ * The <Link> component handles different scenarios of how links are written in the website.
+ * Below are the recommended convention for providing a link `href`. However, to be
+ * as compatible as possible, the component also tries to support other possible scenarios,
+ * including relative filesystem paths, etc.
+ *
+ * ---
+ * # Links conventions
```

Comment:
This is the convention that we agreed on. There will be a proper document as follow up to instruct the writers how to write the links properly.